sdk/tables/azure-data-tables/azure/data/tables/_serialize.py

```diff
@@ -163,14 +166,19 @@ def _to_entity_none(value):  # pylint: disable=unused-argument
     )
 
 # Conversion from Edm type to a function which returns a tuple of the
-# type string and content string.
+# type string and content string. These conversions are only used when the
+# full EdmProperty tuple is specified. As a result, in this case we ALWAYS add
+# the Odatatype tag, even for field types where it's not necessary. This is why
+# boolean and int32 have special processing below, as we would not normally add the
+# Odatatype tags for these to keep payload size minimal.
+# This is also necessary for CLI compatibility.
 _EDM_TO_ENTITY_CONVERSIONS = {
     EdmType.BINARY: _to_entity_binary,
-    EdmType.BOOLEAN: _to_entity_bool,
+    EdmType.BOOLEAN: lambda v: (EdmType.BOOLEAN, v),
```

Comment:
Why it's not `lambda v: (EdmType.BOOLEAN, bool(v))`?

Reply:
This is because `bool(v)` will not always evaluate correctly. For example:
`bool("false") == True`